src/compiler/program.ts

```diff
@@ -1168,9 +1168,6 @@ namespace ts {
                             let typeAssertionExpression = <TypeAssertion>node;
                             diagnostics.push(createDiagnosticForNode(typeAssertionExpression.type, Diagnostics.type_assertion_expressions_can_only_be_used_in_a_ts_file));
                             return true;
-                        case SyntaxKind.Decorator:

```

Comment:
i am just worried of breaking these users later with no warning. the switch is a implicit agreement that you may be broken. not sure if this changes your feelings when you really are broken though.